aws_lambda_powertools/utilities/idempotency/idempotency.py

```diff
@@ -66,174 +57,88 @@ def idempotent(
     """
 
     config = config or IdempotencyConfig()
+    args = event, context
     idempotency_handler = IdempotencyHandler(
-        lambda_handler=handler, event=event, context=context, persistence_store=persistence_store, config=config
+        function=handler,
+        function_payload=event,
+        idempotency_config=config,
+        persistence_store=persistence_store,
+        function_args=args,
+        function_kwargs=kwargs,
     )
 
-    # IdempotencyInconsistentStateError can happen under rare but expected cases when persistent state changes in the
-    # small time between put & get requests. In most cases we can retry successfully on this exception.
-    # Maintenance: Allow customers to specify number of retries
-    max_handler_retries = 2
-    for i in range(max_handler_retries + 1):
-        try:
-            return idempotency_handler.handle()
-        except IdempotencyInconsistentStateError:
-            if i == max_handler_retries:
-                # Allow the exception to bubble up after max retries exceeded
-                raise
+    return idempotency_handler.handle()
 
 
-class IdempotencyHandler:
+def idempotent_function(
+    function: Optional[AnyCallableT] = None,
+    *,
+    data_keyword_argument: str,
```

Comment:
@heitorlessa  - i added `data_keyword_argument` is not part of `ImdempotencyConfig` as it only applies to function decorators? Although even a lambda handler (via a decorator have additional parameters)

Reply:
yes, that's right @michaelbrewer. Because Lambda has a known signature where the `data_keyword_argument` would always be event, so we don't need it.